Makefile

```
@@ -18,3 +18,44 @@ wasm:
 
 parser:
 	go generate .
+
+build:
+	go build -ldflags=$(LDFLAGS) ./cmd/arrai
+
+
+######## Build information report
+originalVersion=$(shell git describe --tags)
+
+ifeq (-, $(findstring -, $(originalVersion))) #it is in branch
+tagName= $(firstword $(subst -,  ,$(originalVersion)))
+diffLogs = $(foreach item, $(shell git log --pretty=format:"%h" $(tagName)..HEAD),$(item))
+
+ifeq (true, $(words $(diffLogs)) = 0 || $(shell git status -s) = ) # no changes
+Version=$(tagName)
+else
+
+ifeq (, $(shell git status -s)) # it doesn't have uncommitted or untrakced changes
+Version=$(FullCommit)~$(words $(diffLogs)) = $(tagName)
+else
+Version=DIRTY-$(FullCommit)~$(words $(diffLogs)) = $(tagName)
+endif
```

This looks like it could use some deduplication:

```suggestion
Version=$(FullCommit)~$(words $(diffLogs)) = $(tagName)
ifneq (, $(shell git status -s)) # there are uncommitted or untracked changes
Version=DIRTY-$(Version)
endif
```

---

I think current code is more readable, and `Version=DIRTY-$(Version)` couldn't work as `Makefile:39: *** Recursive variable 'Version' references itself (eventually).  Stop.`